File: backend/config/venues.py
"""Venue configuration for Warsaw pool venues.

To add a venue, you need:
1. venue_id - from the CueScore URL
2. slug - the URL-friendly name from the CueScore URL
3. name - display name

Example venue URL: https://cuescore.com/venue/147-break-nowogrodzka/67496954
- venue_id: "67496954"
- slug: "147-break-nowogrodzka"
- name: "147 Break Nowogrodzka"

Note: The base venue URL is used. The scraper automatically appends /tournaments
when fetching tournament lists.
"""

# Warsaw venues to track
WARSAW_VENUES = [
    {
        "id": "2842336",
        "slug": "147-break-zamieniecka",
        "name": "147 Break Zamieniecka"
    },
    {
        "id": "2983061",
        "slug": "147-break-fort-wola",
        "name": "147 Break Fort Wola"
    },
    {
        "id": "1698108",
        "slug": "147-break-nowogrodzka",
        "name": "147 Break Nowogrodzka"
    },
    {
        "id": "57050170",
        "slug": "shooters",
        "name": "Shooters"
    },
    {
        "id": "3367445",
        "slug": "eighty-nine",
        "name": "Eighty Nine"
    },
    {
        "id": "36031138",
        "slug": "zlota-bila-centrum-bilardowe",
        "name": "Złota Bila - Centrum Bilardowe"
    },
    {
        "id": "2357769",
        "slug": "billboard-pool-snooker",
        "name": "Billboard Pool & Snooker"
    },
    {
        "id": "1634568",
        "slug": "klub-pictures",
        "name": "Klub Pictures"
    },
    {
        "id": "22253992",
        "slug": "the-lounge-billiards-club",
        "name": "The Lounge - Billiards Club"
    },
]

File: backend/src/config/venues.rs
/// Venue configuration for tournament scraping
///
/// To find venue IDs on CueScore:
/// 1. Navigate to https://cuescore.com
/// 2. Search for the venue by name
/// 3. Click on the venue
/// 4. The URL will be: https://cuescore.com/venue/{name}/{id}/
/// 5. Extract the ID from the URL
#[derive(Debug, Clone)]
pub struct VenueConfig {
    pub id: i64,
    pub slug: &'static str,
    pub name: &'static str,
}

impl VenueConfig {
    pub fn new(id: i64, slug: &'static str, name: &'static str) -> Self {
        Self { id, slug, name }
    }
}

/// Get the list of Warsaw pool venues to scrape for tournaments
pub fn get_venues() -> Vec<VenueConfig> {
    vec![
        VenueConfig::new(2842336, "147 Break Zamieniecka", "147 Break Zamieniecka"),
        VenueConfig::new(29830615, "147 Break Fort Wola", "147 Break Fort Wola"),
        VenueConfig::new(1698108, "147 Break Nowogrodzka", "147 Break Nowogrodzka"),
        VenueConfig::new(57050170, "Shooters", "Shooters"),
        VenueConfig::new(3367445, "Eighty Nine", "Eighty Nine"),
        VenueConfig::new(36031138, "Złota Bila - Centrum Bilardowe", "Złota Bila - Centrum Bilardowe"),
        VenueConfig::new(2357769, "Billboard pool & snooker", "Billboard Pool & Snooker"),
        VenueConfig::new(1634568, "Klub Pictures", "Klub Pictures"),
        VenueConfig::new(22253992, "The Lounge - Billiards Club", "The Lounge - Billiards Club"),
    ]
}
